documentation/API_documentation/Simple_Edge_Discovery_User_story.md

```diff
@@ -2,15 +2,15 @@
- **Scope:** *Order To Activate (OTA)* \- Create/Remove/Get service instance (QoD session resource)\, get notification about the instance (QoD session resource) |
-| ***Pre-conditions*** |The preconditions are listed below:<br><ol><li>The Customer:BusinessManager and Customer:Administrator have been onboarded to the CSP's API platform.</li><li>The Customer:BusinessManager has successfully subscribed to the QoD product from the product catalog.</li><li>The Customer:Administrator has onboarded the Customer:User to the platform.</li><li>The Customer either owns the device subscriptions or has agreed to the terms and conditions of the CSP for managing consent of subscription owners.</li><li>The means to get the access token are known to the Customer:User to ensure secure access of the API.|
+| ***Summary*** | As an application developer belonging to an enterprise, I want to request (using either my application server/backend service, or an HTTP application client on the end user terminal) the closest MEC Platform to a given end user's terminal. |
+| ***Roles, Actors and Scope*** | **Roles:** Customer:User<br> **Actors:** Application service providers, network operators, application developers. The API does not explicitly handle consent management (expects that network information related to proximity of terminal to MEC platforms is not sensitive information ) and hence end users are not included as Actors. Should this change, the actor list will be extended with end users. <br>
+ **Scope:** Get closest MEC platform to a target UE |
+| ***Pre-conditions*** |The preconditions are listed below:<br><ol><li>The Customer:BusinessManager and Customer:Administrator have been onboarded to the CSP's API platform.</li><li>The Customer:BusinessManager has successfully subscribed to the EDS product from the product catalog.</li><li>The Customer:Administrator has onboarded the Customer:User to the platform.</li><li>The Customer:User has obtained a valid identifier for the target UE.</li>|
 
-| ***Activities/Steps*** | **Starts when:** The customer application server|client makes a GET request to the EDS API to query the closest MEC platofrm to the end user terminal. The end user terminal is either implicitly identified (e.g. by its source IP when attached to a cellular network) or explicitly identified (the identity is passed in the request querystring)<br>**Ends when:** The EDS API responds to the customer application server|client . |
+| ***Activities/Steps*** | **Starts when:** The customer application server|client makes a GET request to the EDS API to query the closest MEC platform to the target UE (an end user terminal). The target UE is either implicitly identified (e.g. by its source IP when attached to a cellular network) or explicitly identified (the identity is passed in the request querystring)<br>**Ends when:** The EDS API responds to the customer application server|client . |
```

Comment:
@nicolacdnll The IP assumes in EDS is the public IP. N6, in most of the cases, is the network on a  public DNN (except for private network). Hence N6 IP is uniquely identified and routable. UPF could assign private local IP, to the device, from the common pool on internal network, and perform  NAT'ing for outside interface (N6).

Reply:
Let's see if I understood you: Are you suggesting that on a telco's public DNN, each UE gets an unique IP? Yeah, it makes sense, and I'm probably biased by MPNs.

If that's it, good and thanks. Otherwise, I think I've missed the point.

Reply:
I think the above point is very important, indeed the IP submitted needs to be the *private*, unique IP and not the public (NATted) IP. This needs to be emphasized in the corresponding documentation to developers.

In general @nicolacdnll's suggestion to rather use location data than IP makes a lot of sense as this would remove the need for the network to perform the function of translating the private IP address into a location information. I would though not use network specific location data (i.e. LAC/TAC) but absolute geocordinates which can be obtained by the device using device-specific mechanisms irrespective of the access technology. In that way e.g. also devices using non-3GPP access could be supported.

Reply:
@ThomasEdgeXR Private IPs are not unique as they are re-used across multiple gateways.

Reply:
Exactly my point Syed.
However, if we use the public IP I think that the telco systems need a way to bind/map cloudlets at edge to the DN they are in or to the DN's Public IP. If that is not done, imagining the scenario in the sketch below (oversimplified I know!), how could the system know, and suggest, to the UEs behind the distributed UPF that the _Cloudlet @Edge_ has lower latency than the _Cloudlet @Core_?

![image](https://github.com/camaraproject/EdgeCloud/assets/1780330/1e5618fe-6c56-41b6-866d-c7b30f4369e8)

Reply:
Also, with public IP in telco network, the binding of an public IP:Port may change with time and can be assigned to other users if the current user becomes Inactive for any reason e.g. device get powered off. So there could be a chance that information become stale which may provide incorrect information at times. So the reliability shall be ensured while defining the behavior.

Reply:
> Exactly my point Syed. However, if we use the public IP I think that the telco systems need a way to bind/map cloudlets at edge to the DN they are in or to the DN's Public IP. If that is not done, imagining the scenario in the sketch below (oversimplified I know!), how could the system know, and suggest, to the UEs behind the distributed UPF that the _Cloudlet @edge_ has lower latency than the _Cloudlet @core_?

@nicolacdnll good point. However, i think it would depend on a operator specific implementation how to define the closest edge. The closest edge could be a lowest latency candidate or it could be a closest hop w.r.t to UE's anchor point.

Reply:
> Also, with public IP in telco network, the binding of an public IP:Port may change with time and can be assigned to other users if the current user becomes Inactive for any reason e.g. device get powered off. So there could be a chance that information become stale which may provide incorrect information at times. So the reliability shall be ensured while defining the behavior.

@gunjald This could be addressed by exposing a notification endpoint for IP changes due to mobility or other factors.

Reply:
> @nicolacdnll good point. However, i think it would depend on a operator specific implementation how to define the closest edge. The closest edge could be a lowest latency candidate or it could be a closest hop w.r.t to UE's anchor point.

Sounds fair, but if that's the case I would suggest we add this consideration to the user story or assumptions.

Reply:
> > Also, with public IP in telco network, the binding of an public IP:Port may change with time and can be assigned to other users if the current user becomes Inactive for any reason e.g. device get powered off. So there could be a chance that information become stale which may provide incorrect information at times. So the reliability shall be ensured while defining the behavior.
> 
> @gunjald This could be addressed by exposing a notification endpoint for IP changes due to mobility or other factors.

Similar to the other point, wouldn't it make sense that it's up to each operator implementation to keep track of the mapping of each UE's <privateIP->publicIP:port>? Unless there is a real use case, this mapping should be completely transparent for UE's and developers.